activerecord/test/cases/connection_adapters/merge_and_resolve_default_url_config_test.rb

```diff
@@ -273,6 +273,41 @@ def test_merge_conflicts_with_database_url
                     }
         assert_equal expected, actual
       end
+
+      def test_merge_no_conflicts_with_database_url_and_adapter
+        ENV["DATABASE_URL"] = "postgres://localhost/foo"
+
+        config   = { "default_env" => { "adapter" => "postgresql", "pool" => "5" } }
+        actual   = resolve_config(config)
+        expected = { "default_env" =>
+                     { "adapter"  => "postgresql",
+                       "database" => "foo",
+                       "host"     => "localhost",
+                       "pool"     => "5"
+                     }
+        }
+        assert_equal expected, actual
+
+        ENV["DATABASE_URL"] = nil
```

Comment:
I think this is already done in [`teardown`](https://github.com/rails/rails/pull/36565/files#diff-0e21b63c5ed26f357a66d05a4c6a3804R15)

Reply:
Yes I accidentally copied that from the script sent by the original reporter. Fixing now.